CSS Variables: Refactor footer sizing and alignment

machikoyasuda opened the pull request:
closes #974 

## What this PR does
- Fixes: Fixes the current issue where on Desktop, the footer is not neatly at the bottom of the page on first load for pages that don't fill up the full viewport height.
- Redesigns: The footer height is now 50px on Desktop. The footer is slightly taller now on Desktop.
- Fixes: The type (leter-spacing, font-weight) for the footer was off. Now it's not.
- Deletes all old CSS code that tried to place the footer.
- Moves around CSS styles into categories and comments them.

## How to test
- Mobile: Footer should have no regressions
- Desktop: Go through short pages (agency index) - Footer should be at the bottom. 
- Desktop: Go through long pages (help, eligibility start) - Footer should have no regressions.
- Test the app in Private mode or a different browser to test the `visited` color logic.

## Nitty gritty details
- Removes `.global-footer` so this footer is now 100% free of custom CA Web Std Template styles. We were overriding most of them. Uses bootstrap classes instead to declare most styles. Uses CSS `gap` to separate the links on Desktop.
- Moves all footer-related CSS variables to right above the footer element CSS styles. We can undo this, but this helped me write/read this code.
- Logic for the viewport height, explained: This means, "Set the _minimum_ height of the center div to 100% of the viewport height, minus the height of the footer and the header". That means, if the natural height of the page is _more than_ 100vh from the start (which would apply to all pages on a smaller mobile device and some pages on a Desktop device), then the footer will flow naturally. **Edited to add:** The footer is sticky but not fixed. Fixed would mean the footer is always there even after you scroll.
```css
--main-content-min-height: calc(100vh - 182px);
  /* 182px = Header Height (80px) + (Footer Link (50px) * Number of Links (2)) + Underline Height (2px) */
--main-content-min-height: calc(100vh - 130px);
/* 130px = Header Height (80px) + Footer Height (50px) */
```

## Post merge actions:

After this PR is merged and deployed on Development, we should ask @srhhnry to test the footer out on her mobile and desktop devices.

## Screenshots

Short page - first load
<img width="1510" alt="image" src="https://user-images.githubusercontent.com/3673236/193887616-183883a5-8fc7-4a3c-9e99-34dde1125bd7.png">

Short page - visited links
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/3673236/193887563-af988254-1ed8-4686-a9e6-14a3ade0eb52.png">

Long page - first load
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/3673236/193887486-003ce60a-3492-41b4-a821-1c40bee2480b.png">

Mobile - 
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/3673236/193887444-9b5b826f-9f66-49ac-86cb-07bebe504f65.png">

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/3673236/193887410-0d35e5fc-b88a-486d-b588-d7f7e0bb9565.png">

angela-tran commented:
> The footer is not sticky. Sticky would mean the footer is always there even after you scroll.

Just want to note this to improve our communication: I think we're using this term to mean two different things. I understood it to mean that the footer sticks to the bottom of the viewport if the page content is not enough to push it down. If there's enough content, then it will push the footer down. There is some MDN documentation on this: https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Sticky_footers

The "Hey!" note on https://css-tricks.com/couple-takes-sticky-footer/ also tries to clarify this

machikoyasuda commented:
@angela-tran Oooh yes ~ It is sticky but _not_ **fixed**.

Just added this up top

>  **Edited to add:** The footer is sticky but not fixed. Fixed would mean the footer is always there even after you scroll.

thekaveman commented on benefits/static/css/styles.css:
```diff
@@ -143,61 +143,115 @@ h3,
 }
 
-/* class styles */
+@media (max-width: 992px) {
```
Can you please call out here with a comment that this is a non-mobile-first media query? And maybe elsewhere too if needed... will help us go back and clean up later.

machikoyasuda replied:
I will add a comment into the code and also put it here: This is because of the way the Footer on Mobile _must_ have 100% width fully to get the full-width white line. And the Footer on Desktop+ _needs_ the Bootstrap `.container` logic to size its widths. If we removed the `.container` and went with `width: 100%` (to be mobile-first), then we'd have to re-write our own container width logic, which would be very difficult. So basically, it's because of the full-width line that goes edge to edge.

thekaveman replied:
Makes sense and that is totally fine! A little doc comment describing that would be helpful is all.

thekaveman commented on benefits/static/css/styles.css:
```diff
@@ -566,36 +584,6 @@ footer.global-footer .footer-links a:active {
 }
 
 @media (max-width: 767px) {
```
same comment about non-mobile-first media query

machikoyasuda replied:
These ones and lower are because I have not had a chance to work on radio buttons yet. All the media queries below this are just stuff I have not gotten to.

thekaveman commented on benefits/core/templates/core/base.html:
```diff
@@ -71,14 +71,14 @@
       {% endblock main_content %}
     </main>
 
-    <footer id="footer" class="global-footer">
```
Losing this class doesn't break anything from the State Template, right?

machikoyasuda replied:
As far as I've tested no.

thekaveman reviewed (dismissed):
Few minor comment requests

angela-tran commented on benefits/static/css/styles.css:
```diff
@@ -32,12 +28,16 @@
     --h1-font-size: calc(35rem / 16);
     --h1-text-align: center;
+    --main-content-min-height: calc(100vh - 130px);
+    /* 130px = Header Height (80px) + Footer Height (50px) */
```
This line was initially inside the `@media (min-width: 767px)` media query at b35e375d8a1b14dfab5a6e925da26f44494413dd, but then the media query moved in https://github.com/cal-itp/benefits/pull/1014/commits/10cbca03a5230dffb5528e5a7890c86da1031827#diff-744df555aee38173a0b0868baa7befa720e6a28f80747523e74e724c126cd381L46-L47 without taking this line. I can't tell what the intended change was, so could you clarify?

machikoyasuda replied:
https://github.com/cal-itp/benefits/pull/1014/commits/f9b605fbd115639ee5c4a4301ae590ae951378d7

Footer breakpoint is 992px (can be changed if Design wants to change it)

I want Design to test the footer on dev-benefits.calitp.org on real devices before we change the breakpoint though.

machikoyasuda commented:
@thekaveman @angela-tran After this is approved, I can take care of the conflict.
